### PR TITLE
Don't use the same memoized function across instances when `.decorator()` is used

### PR DESCRIPTION
**Pull request by Richienb**
Fixes: #74

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -204,7 +204,14 @@ Returns a TypeScript decorator which memoizes the given function.
 
 Type: `object`
 
-Same as options for `mem()`.
+Same as options for `mem()` and the following:
+
+##### isAcrossInstances
```
Comment (fregante) [inline readme.md:209]: When would this ever be useful? 🤷‍♂️ Instances are not meant to share a global states

Comment (fregante) [reply readme.md:209]: The only situation that we're not covering anywhere are static methods (isAcrossInstances doesn't apply here either)

Comment (sindresorhus) [reply readme.md:209]: @fregante It's not something you would need often, but sometimes you do calculations that don't actually depend on any state in the instance. I don't have an immediate need for this though, just thought someone might find it useful. Do you think it's worth keeping it?

Comment (fregante) [reply readme.md:209]: > sometimes you do calculations that don't actually depend on any state in the instance

Isn't that the definition of a static method?

Comment (sindresorhus) [reply readme.md:209]: Do decorators work on static methods? If so, we can indeed remove this option.

Comment (fregante) [reply readme.md:209]: The docs don't include any static examples, but it doesn't mention "static" a few times: https://www.typescriptlang.org/docs/handbook/decorators.html#decorator-evaluation

Also it seems to pass:

https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABMGAnAzlAFASkQbwChFEIF04AbAUwDpK4BzLAIhQ2xwC5kBDaOKgCeiagDdelELyjUAJixwBuYolTUoIVElCRYCRFii9UjDT15ghAGkQAHVHDvVUUIQGlqQnplQwwjLZy1OgQfnZQgjwACo7OrkIAIiFhMBGCeEQkJGRgFDT0TKzsmLg8EJI0CsqqAL4qtYSEEJS86OiIAOLqGi4EqgACJZyqjD2c-dlqGlpILAAS1JQMLCokjSSDw7iqmDIwEIhj1Bq4k9nqmtqIC0sra4iNtUA

Screen Shot

Comment (sindresorhus) [reply readme.md:209]: Alright let's remove the option and update the docs to mention support for static methods too, just to be user-friendly.

Comment (fregante) [reply readme.md:209]: @Richienb I just merged the other PR that extended the docs for this. Update this branch before extending the docs further.

File: index.ts [hunk 2]
```diff
@@ -141,7 +143,7 @@ const mem = <
 export = mem;
 
 /**
-@returns A TypeScript decorator which memoizes the given function.
+@returns A [decorator](https://github.com/tc39/proposal-decorators) to memoize class methods or static class functions.
```
Comment (sindresorhus) [inline index.ts:146]:
```suggestion
@returns A [decorator](https://github.com/tc39/proposal-decorators) to memoize class methods or static class methods.
```

**Comment by sindresorhus** [comment]
👍

**Pull request merged**